cognite/extractorutils/extraction_pipelines.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from functools import wraps
+from threading import Event, Thread
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import ExtractionPipelineRun
+
+
+def add_extraction_pipeline(
+    extraction_pipeline_ext_id: str,
+    cognite_client: CogniteClient,
+    logger: logging.Logger = None,
+    heartbeat_waiting_time: int = 5,
+    added_message: str = "",
+):
+    """
+    This is to be used as a decorator for extractor functions to add extraction pipeline information
+
+    Args:
+        extraction_pipeline_ext_id:
+        cognite_client:
+        logger:
+        heartbeat_waiting_time:
+        added_message:
+
+    Usage:
+        If you have a function named "extract_data(*args, **kwargs)" and want to connect it to an extraction
+        pipeline, you can use this decorator function as:
+        @add_extraction_pipeline(
+            extraction_pipeline_ext_id=<INSERT EXTERNAL ID>,
+            cognite_client=<INSERT COGNITE CLIENT OBJECT>,
+            logger=<INSERT LOGGER>,
+        )
+        def extract_data(*args, **kwargs):
+            <INSERT FUNCTION BODY>
+
+    Potential Improvements:
+    -- Refactor so this decorator share methods with the Extractor context manager in .base.py as they serve a similar
+    purpose
+    -- Change 'cognite_client.extraction_pipeline_runs' -> 'cognite_client.extraction_pipelines.runs'
+    when SDK is updated
```

Comment:
Very good points for further improvements! But I suggest putting them in a normal comment with a `TODO` prefix instead of in the docstring as it is not particularly relevant to a user of the decorator :)

Reply:
Absolutely! Will fix.